pinecone-io/image-search-example
# Add image upsert and delete functionality

## Problem

Describe the purpose of this change. What problem is being solved and why?

_Add image upsert and delete functionality. This change demonstrates Pineconeʼs data freshness (new images are immediately queryable) and the SDKʼs ease of use._

## Solution

Describe the approach you took. Link to any relevant bugs, issues, docs, or other resources.

_Add two new endpoints, `/uploadImages` and `/deleteImage`. Add a UI for the user to upload and delete images and their associated vector embeddings._

## Type of Change


## Test Plan

Describe specific steps for validating this change.

_The following manual tests were performed successfully:_

- _Index the dataset from scratch;_
- _Index the dataset when an index already exists;_
- _Upload a new image;_
- _Click new and existing images (verify relevant results are returned);_
- _Click "Next" and "Previous" buttons (verify the pagination works); and_
- _Delete existing images._
